tools/auto_fill_inference_cache.py

```diff
@@ -111,23 +203,39 @@ def compile_and_cache_model(hf_model_id, batch_size, sequence_length, num_cores,
 if __name__ == "__main__":
     parser = argparse.ArgumentParser(description="Compile and cache a model to the Hugging Face Hub.")
     parser.add_argument("--hf_model_id", type=str, help="Hugging Face model ID to compile.")
+    parser.add_argument(
```

Comment:
I think we could deduce this from the model_id.

Reply:
yeah, theoretically we can leverage `TasksManager.infer_task_from_model` but it's not working for now for stable diffusion models, I made a pull request in Optimum: https://github.com/huggingface/optimum/pull/1793. I can perhaps put a workaround here before it makes its way to a stable optimum release.

Reply:
I think that we need to at least specify in the config files that the task is "text-generation" for decoders if we want to keep just one script for encoder / decoder and stable diffusion. Event with the TaskManager, the default task inferred from eg. gpt2 is feature-extraction.

Reply:
I don't think so. The neuronx exporter code works without specifying a task AFAIK, even for gpt2.
https://github.com/huggingface/optimum-neuron/blob/3005c77071bd9a132c1509355bec34546daedede/optimum/exporters/neuron/__main__.py#L561
To figure out the model is a decoder, I get the task, then the config, all from the model_id.

Reply:
Are you sure? Without specifying the task `text-generation`, I am getting the following error when trying to export decoder:

```bash
optimum-cli export neuron --model hf-internal-testing/tiny-random-gpt2 --sequence_length 64 --batch_size 1 --auto_cast_type bf16 --num_cores 1 tiny_gpt
```

```
Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/ubuntu/pyvenv/aws_neuron_venv2.18_pt212/lib/python3.8/site-packages/optimum/exporters/neuron/__main__.py", line 616, in <module>
    main()
  File "/home/ubuntu/pyvenv/aws_neuron_venv2.18_pt212/lib/python3.8/site-packages/optimum/exporters/neuron/__main__.py", line 567, in main
    input_shapes, neuron_config_class = get_input_shapes_and_config_class(task, args)
  File "/home/ubuntu/pyvenv/aws_neuron_venv2.18_pt212/lib/python3.8/site-packages/optimum/exporters/neuron/__main__.py", line 119, in get_input_shapes_and_config_class
    neuron_config_constructor = TasksManager.get_exporter_config_constructor(
  File "/home/ubuntu/pyvenv/aws_neuron_venv2.18_pt212/lib/python3.8/site-packages/optimum/exporters/tasks.py", line 1961, in get_exporter_config_constructor
    raise ValueError(
ValueError: gpt2 doesn't support task feature-extraction for the neuron backend. Supported tasks are: text-generation.
Traceback (most recent call last):
  File "/home/ubuntu/pyvenv/aws_neuron_venv2.18_pt212/bin/optimum-cli", line 8, in <module>
    sys.exit(main())
  File "/home/ubuntu/pyvenv/aws_neuron_venv2.18_pt212/lib/python3.8/site-packages/optimum/commands/optimum_cli.py", line 163, in main
    service.run()
  File "/home/ubuntu/pyvenv/aws_neuron_venv2.18_pt212/lib/python3.8/site-packages/optimum/commands/export/neuronx.py", line 266, in run
    subprocess.run(full_command, shell=True, check=True)
  File "/usr/lib/python3.8/subprocess.py", line 516, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command 'python3 -m optimum.exporters.neuron --model hf-internal-testing/tiny-random-gpt2 --sequence_length 64 --batch_size 1 --auto_cast_type bf16 --num_cores 1 tiny_gpt' returned non-zero exit status 1.
```

Reply:
For gpt2 it works, but not for the tiny model.